py-polars/polars/series/utils.py

```diff
@@ -83,11 +73,19 @@ def _expr_lookup(namespace: str | None) -> set[tuple[str | None, str, tuple[str,
             if callable(m):
                 # add function signature (argument names only) to the lookup
                 # as a _possible_ candidate for expression-dispatch
+                m = _undo_decorators(m)
                 args = m.__code__.co_varnames[: m.__code__.co_argcount]
                 lookup.add((namespace, name, args))
     return lookup
 
 
+def _undo_decorators(function: Callable[P, T]) -> Callable[P, T]:
```

Comment:
Minor naming suggestion; `_undo_decorators` doesn't really undo the decorators so much as get you the undecorated/root object - how about just `_undecorated` instead?

Reply:
Fair enough - someone else had the same thought here:
https://pypi.org/project/undecorated/

Renamed!